bin/propolis-server/src/lib/initializer.rs

```diff
@@ -1112,6 +1112,28 @@ impl MachineInitializer<'_> {
         smb_tables.commit()
     }
 
+    fn generate_e820(&self) -> Result<Entry, MachineInitError> {
+        info!(self.log, "Generating E820 map for guest address space");
+
+        let mut e820_table = fwcfg::formats::E820Table::new();
+
+        for (addr, len, kind) in self.machine.map_physmem.mappings().into_iter()
+        {
+            let addr = addr.try_into().context("usize should fit into u64")?;
```

Comment:
OVMF doesn't care about entries <4GiB, but I feel like it's slightly more useful to everyone to report lowmem ranges here anyway. I want to refresh myself on how PCIe MMIO ranges work out but given that I've run a guest with an NVMe disk it's presumably not _wrong_.